ISSUE-LABELLING.md

```diff
@@ -0,0 +1,234 @@
+planning, and ultimately, reduce friction to keep the libp2p community engaged,
+vibrant, and excited about what's to come.
+
+We are fans of iterative experimentation. In the next weeks/months, as we roll
+this out and adjust it, we will gather feedback and input from stakeholders
+across the ecosystem.
+
+Our endgame is to converge on a system that pleases all language implementers,
+so that we can normalise this taxonomy across all language repos, and plan
+features and changes in total inter-alignment. We are convinced that by creating
+a cohesive and seamless experience across libp2p implementations, we'll unlock
+tremendous value for our beloved downstream users and developers.
+
+## Descriptive labels
+
+These labels describe some aspect of the issue itself. They can generally be set
+when the issue is first filed, or as discoveries are made. We will adopt tools
+that allow submitters to set these labels.
+
+### Functional area (`area/`)
+
+> Defines the functional area of the libp2p stack an
+> issue, report, question, etc. addresses. These labels refer to abstraction
+> layers; to contextualise the issue to a specific implementation, also add a
+> "component" label.
+
+* area/conn-manager
+* area/connection-bootstrapping
+* area/nat-traversal
+* area/peer-routing
+* area/content-routing
+* area/discovery
+* area/identity
+* area/multiplexing
+* area/negotiation
+* area/peerstore
+* area/pubsub
+* area/relay
+* area/security-channel
+* area/transport
+* area/universal
+
+### Component (`comp/`)
+
+> Specifies the actual component (implementation) an issue refers to. Used when
+> it's clear that an issue affects one or many specific components.
+
+* comp/floodsub
+* comp/gossipsub
+* comp/kad-dht
+* comp/noise
+* comp/quic
+* comp/secio
+* comp/tcp
+* comp/tls
+* comp/webrtc
+* comp/wss
+* comp/yamux
+
+### Difficulty (`dif/`)
+
+> Subjectively estimates the difficulty of resolution of an issue, based on
+> prior expertise about libp2p and its constructs, p2p networking, systems
+> programming, etc.
+
+Values and calibration guidance:
+
+* dif/trivial: Can be confidently tackled by newcomers, who are widely
+  unfamiliar with libp2p.
+* dif/easy: A prexisting libp2p user should be able to pick this up.
+* dif/medium: Prior experience in having developed libp2p modules is likely
+  helpful.
+* dif/hard: Suggests that having worked on the specific component affected
+  by this issue is important.
+* dif/expert: Requires extensive knowledge of the history, implications,
+  ramifications of the issue, as well as deep understanding of the libp2p stack
+  and experience with systems and network programming.
+
+### Size (`size/`)
+
+> Estimates the amount of work. This axis is distinct to the complexity, e.g.
+> something can be easy but require a lot of work, or vice versa. 
+
+Values and calibration guidance:
+
+* size/XS: minutes.
+* size/S: several hours.
+* size/M: one day.
+* size/L: 2-3 days.
+* size/XL: 1 week.
+* size/XXL: > 1 week.
```

Comment:
* time/minutes
* time/hours
* time/day
* time/days
* time/week
* time/weeks

(removes a layer of indirection)

Reply:
I dig this, with the only caveat that I'd use the term `effort` instead of `time` (e.g. 3 hours of effort can span 3 days, at the rate of 1hr/day).